ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -390,16 +390,6 @@ public void commit (boolean report_heuristics) throws HeuristicMixed,
 			}
 
 			destroyAction();
-		} else if (outcome == ActionStatus.ABORTED || outcome == ActionStatus.H_ROLLBACK) {
-			throw new TRANSACTION_ROLLEDBACK(ExceptionCodes.FAILED_TO_COMMIT,
```

Comment:
This is what I wanted to ask as the commit is a bit different that we discussed on forums. I noticed that we have duplicated code in both commit and rollback methods - it caused problems in rollback, but not in commit - anyway I removed the last else from commit too, because to my understanding this check is already performed in switch below with identical exceptions beeing thrown.

BTW: In rollback there were no difference between different statuses in if - those differences are present in switch though, so I removed last else instead of adding another if.